tests/unit/utils/test_vt.py

```diff
@@ -212,3 +273,128 @@ def test_isalive_while_theres_data_to_read(self):
+
+    def test_split_multibyte_characters_unicode(self):
+        '''
+            Tests that the vt correctly handles multibyte characters that are
+            split between blocks of transmitted data.
+        '''
+        if not stdout_fileno_available() and 'TEST_VT_CHILD' not in os.environ:
+            self.run_test_in_subprocess(
+                'test_split_multibyte_characters_unicode'
+            )
+            return
+        block_size = 1024
+        encoding = 'utf-8'
+        try:
+            tempdir = tempfile.mkdtemp(suffix='test_vt_unicode')
+            file_path_stdout = os.path.join(tempdir, "vt_unicode_stdout.txt")
+            file_path_stderr = os.path.join(tempdir, "vt_unicode_stderr.txt")
+            stdout_content = b'\xE2\x80\xA6' * 4 * block_size
+            # stderr is offset by one byte to guarentee a split character in
+            # one of the output streams
+            stderr_content = b'\x2E' + stdout_content
+            with salt.utils.fopen(file_path_stdout, "wb") as fout:
+                fout.write(stdout_content)
+            with salt.utils.fopen(file_path_stderr, "wb") as ferr:
+                ferr.write(stderr_content)
+
+            expected_stdout = salt.utils.stringutils.to_unicode(stdout_content,
+                                                                encoding)
+            expected_stderr = salt.utils.stringutils.to_unicode(stderr_content,
+                                                                encoding)
+            python_command = '\n'.join((
+                'import sys',
+                'import os',
+                'with open(\'' + file_path_stdout + '\', \'rb\') as fout:',
+                '    os.write(sys.stdout.fileno(), fout.read())',
+                'with open(\'' + file_path_stderr + '\', \'rb\') as ferr:',
+                '    os.write(sys.stderr.fileno(), ferr.read())',))
+            term = salt.utils.vt.Terminal(
+                args=[sys.executable,
+                      '-c',
+                      '"' + python_command + '"'],
+                shell=True,
+                stream_stdout=False,
+                stream_stderr=False,
+                force_receive_encoding=encoding)
+            buffer_o = buffer_e = salt.utils.stringutils.to_unicode('')
+            try:
+                while term.has_unread_data:
+                    stdout, stderr = term.recv(block_size)
+                    if stdout:
+                        buffer_o += stdout
+                    if stderr:
+                        buffer_e += stderr
+
+                self.assertEqual(buffer_o, expected_stdout)
+                self.assertEqual(buffer_e, expected_stderr)
+            finally:
+                term.close(terminate=True, kill=True)
+        finally:
+            # Ignore errors because on some platforms the tempdir doesn't exist
+            # which would seem be indicative of deeper problems
+            shutil.rmtree(tempdir, ignore_errors=True)
```

Comment:
It would probably be better to use `from tests.support.helpers import with_tempdir` and use the `@with_tempdir` decorator here.

Reply:
Thanks for the suggestion, that is a neater way of doing it so I've switched to doing that.
I'm currently looking into making it not use the temp files since some of the testing environments seem to be missing the temporary directory so creating the temp directory fails.
I had originally not used the temp-files but found that they were the most reliable way to be sure I got exactly the same stream of bytes on all platforms.
I'm not sure if the missing temp dir a transient error so I'll try rerunning the tests.